### PR TITLE
Create Emoji service and implement text parser feature

### PR DESCRIPTION
**Pull request by KonradSzwarc**
- Create an emoji.json generator that fetches emoji data from four data sources and merges them.
- Create a compressor that handles emoji.json minification and de-minification.
- Add the `initialize` method that dynamically loads the emoji.json content into the service.
- Add the `parseText` method that transforms emoji texts and symbols into emoji icons.
- Create GitHub Actions workflow to automatically publish the package when changes are merged to main.

### CODE REVIEW

File: src/generator/data-getter/emoji-base.data-getter.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+   * keyed by the skin tone. For emoji that support multiple skin tones within
+   * a single emoji, each skin tone is separated by a dash character.
+   */
+  skin_variations?: string;
+}
+
+class BaseEmojisDataGetter implements DataGetter {
+  private url = 'https://raw.githubusercontent.com/iamcal/emoji-data/master/emoji.json';
+
+  getData() {
```
Comment (pr0xyMity) [inline src/generator/data-getter/emoji-base.data-getter.ts]: Lets try to use async/await from now on, it's clear and I'd like it to be the standard in our project ;)

File: src/generator/data-getter/emoji-to-keywords.data-getter.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import axios from 'axios';
+import { DataGetter } from './data-getter.types';
+
+type EmojiToKeywordsMap = Record<string, string[]>;
+
+class EmojiToKeywordsDataGetter implements DataGetter {
+  private url = 'https://raw.githubusercontent.com/muan/emojilib/main/dist/emoji-en-US.json';
+
+  getData() {
```
Comment (pr0xyMity) [inline src/generator/data-getter/emoji-to-keywords.data-getter.ts]: Same here ;)

File: src/generator/data-getter/emoji-base.data-getter.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
+import axios from 'axios';
+import { DataGetter } from './data-getter.types';
+
+interface EmojiBase {
```
Comment (pr0xyMity) [inline src/generator/data-getter/emoji-base.data-getter.ts:5]: Did you done this documentation alone?

**Review by pr0xyMity — COMMENTED** [review]
Great code, some small questions

File: src/generator/data-getter/emoji-base.data-getter.ts (hunk 3 above)
Comment (KonradSzwarc) [reply src/generator/data-getter/emoji-base.data-getter.ts:5]: No, it's from the repository we take emojis data from. I would annotate it in the comment

**Review by tgrygiel — APPROVED** [review]

**Pull request merged**